pkg/cmd/cli/cmd/process.go

```diff
@@ -271,19 +275,19 @@ func RunProcess(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []
 }
 
 // injectUserVars injects user specified variables into the Template
-func injectUserVars(values []string, out io.Writer, t *templateapi.Template) {
+func injectUserVars(values []string, t *templateapi.Template) error {
 	for _, keypair := range values {
 		p := strings.SplitN(keypair, "=", 2)
 		if len(p) != 2 {
-			fmt.Fprintf(out, "invalid parameter assignment in %q: %q\n", t.Name, keypair)
-			continue
+			return fmt.Errorf("invalid parameter assignment in %q: %q\n", t.Name, keypair)
 		}
 		if v := template.GetParameterByName(t, p[0]); v != nil {
 			v.Value = p[1]
 			v.Generate = ""
 			template.AddParameter(t, *v)
 		} else {
-			fmt.Fprintf(out, "unknown parameter name %q\n", p[0])
+			return fmt.Errorf("unknown parameter name %q\n", p[0])
```

Comment:
I'm not well-versed here - what are valid use-cases where `oc process` is used and parameters are *not* meant to be updated?

I don't see, either, a use-case where I am giving some parameters into `oc process` that don't belong and everything is OK - if this is used as part of a script, which I assume a lot of  `oc process | oc create -f -` would be, the output to `stderr` won't be as obvious and the resulting objects in etcd from the `create` may not be correct.